src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -113,11 +114,55 @@ internal static bool IsAllNullBytes(Span<byte> buffer)
             return true;
         }
 
-        // Returns a DateTimeOffset instance representing the number of seconds that have passed since the Unix Epoch.
-        internal static DateTimeOffset GetDateTimeFromSecondsSinceEpoch(double secondsSinceUnixEpoch)
+        // Converts the specified number of seconds that have passed since the Unix Epoch to a DateTimeOffset.
+        internal static DateTimeOffset GetDateTimeOffsetFromSecondsSinceEpoch(double secondsSinceUnixEpoch) =>
+            new DateTimeOffset((long)(secondsSinceUnixEpoch * TimeSpan.TicksPerSecond) + DateTime.UnixEpoch.Ticks, TimeSpan.Zero);
+
+        // Converts the specified DateTimeOffset to the number of seconds that have passed since the Unix Epoch.
+        internal static double GetSecondsSinceEpochFromDateTimeOffset(DateTimeOffset dateTimeOffset) =>
+            ((double)(dateTimeOffset.UtcDateTime - DateTime.UnixEpoch).Ticks) / TimeSpan.TicksPerSecond;
+
+        // If the specified fieldName is found in the provided dictionary and it is a valid double number, returns true and sets the value in 'dateTimeOffset'.
+        internal static bool TryGetDateTimeOffsetFromTimestampString(IReadOnlyDictionary<string, string> dict, string fieldName, out DateTimeOffset dateTimeOffset)
+        {
+            dateTimeOffset = default;
+            if (dict.TryGetValue(fieldName, out string? value) &&
+                double.TryParse(value, NumberStyles.Any, CultureInfo.InvariantCulture, out double secondsSinceEpoch))
```

Comment:
`double.TryParse` does some unexpected things.  Like "987,3" is 9873 (as if there were no comma).  Maybe that's OK, maybe it's not, but thought I'd point it out.

Reply:
Thanks, yes. That comma was giving me a headache on the first PR. I fixed that problem by adding the `NumberStyles.Any, CultureInfo.InvariantCulture` arguments.

Reply:
What are you expecting if the input has that comma, though?

```C#
        private static void Main()
        {
            const string Test = "123,45";
            Console.WriteLine(Test);
            if (double.TryParse(Test, NumberStyles.Any, CultureInfo.InvariantCulture, out double secondsSinceEpoch))
            {
                Console.WriteLine($"True: {secondsSinceEpoch}");
            }
            else
            {
                Console.WriteLine(false);
            }
        }
```

produces

```
123,45
True: 12345
```

Reply:
If the input has such comma, it's malformed. The spec says it should have a dot as separator. Should I add logic to handle such unexpected numbers?